src/parser/shaman/elemental/modules/talents/Ascendance.js

```diff
@@ -35,36 +52,94 @@ class Ascendance extends Analyzer {
     return this.rawUpdate;
   }
 
+  get averageLavaBurstCasts() {
+    return this.numCasts[SPELLS.LAVA_BURST.id]/this.numCasts[SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id];
+  }
+
   on_byPlayer_cast(event) {
-    if (this.selectedCombatant.hasBuff(SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id, event.timestamp)) {
-      const spellId = event.ability.guid;
-      if (this.numCasts[spellId] !== undefined) {
-        this.numCasts[spellId] += 1;
-      } else {
-        this.numCasts.others += 1;
+    const spellId = event.ability.guid;
+    const target = this.enemies.getEntity(event);
+
+
+    if(spellId === SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id) {
+      this.justEnteredAcendance = true;
+      this.numCasts[SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id]++;
+    }
+
+    if(this.justEnteredAcendance){
+      this.justEnteredAcendance=false;
+      if(target && target.getBuffUptime(SPELLS.FLAME_SHOCK.id) < ASCENDANCE_DURATION){
+        this.badFSAscendence++;
       }
     }
+
+
+    const ability = this.abilities.getAbility(spellId);
+    if(!ability){
+      return;
+    }
+    if(!this.selectedCombatant.hasBuff(SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id, event.timestamp) || spellId === SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id){
+      return;
+    }
+
+    const gcd = this._resolveAbilityGcdField(ability.gcd);
```

Comment:
Why is this here? To ignore spells not on GCD? Why aren't they supposed to be counted as `others`?

Reply:
I basically just want to check if you wasted any gcds on spells other then lvb+eb+es.

If you cast anything that is not on the gcd, like trinkets or racials or whatnot, it shouldn't yell at you.